src/main/kotlin/org/ionproject/integration/step/tasklet/iseltimetable/implementations/DownloadAndCompareTasklet.kt

```diff
@@ -0,0 +1,56 @@
+package org.ionproject.integration.step.tasklet.iseltimetable.implementations
+
+import java.io.File
+import java.net.URI
+import org.ionproject.integration.file.implementations.FileDownloaderImpl
+import org.ionproject.integration.file.implementations.PDFBytesFormatChecker
+import org.ionproject.integration.step.tasklet.iseltimetable.exceptions.DownloadAndCompareTaskletException
+import org.ionproject.integration.utils.Try
+import org.ionproject.integration.utils.orThrow
+import org.springframework.batch.core.StepContribution
+import org.springframework.batch.core.configuration.annotation.StepScope
+import org.springframework.batch.core.scope.context.ChunkContext
+import org.springframework.batch.core.step.tasklet.Tasklet
+import org.springframework.batch.repeat.RepeatStatus
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.PropertySource
+import org.springframework.stereotype.Component
+
+@Component(value = "DownloadAndCompareTasklet")
+@StepScope
+@PropertySource("isel-timetable.properties")
+class DownloadAndCompareTasklet() : Tasklet {
+
+    @Value("\${pdf-remote-location}")
```

Comment:
Same.

Reply:
Which is the config source for these values?

Reply:
Isolated all the configs for the job in one class, and added coercion from string type to more appropriate types.